docs/development/debugging-in-goland.md

```diff
@@ -0,0 +1,132 @@
+# Debugging in GoLand
+
+<!-- panels:start -->
+
+<!-- div:right-panel -->
+
+![GoLand icon](../_images/icon-goland.svg)
+
+> [JetBrains GoLand](https://www.jetbrains.com/go/) is an IDE for the Go
+> programming language.
+
+<!-- div:left-panel -->
+
+Running Wharf locally is a hassle, as every code change requires you to rebuild
+the entire Docker images, and you cannot connect a debugger.
+
+It is possible to hook up [JetBrains GoLand](https://www.jetbrains.com/go/) to
+accomplish this.
+
+?> To those that have experience with the container tools in Visual Studio will
+   feel right at home.
+
+<!-- panels:end -->
+
+## 1. Install GoLand
+
+Download GoLand from this link: <https://www.jetbrains.com/go/download>
+
+## 2. Open project you want to debug
+
+Let's say it's the main API, then you open the `api` project using GoLand.
+
+## 3. Modify the Dockerfile
+
+1. In first step of multistage build we need to install Delve debugger:
+   <https://github.com/go-delve/delve>
+
+2. Set additional flags in `go build` command: `-gcflags "all=-N -l"`
+
+3. In second step of multistage build use `debian:buster` image as base
+   image. `scratch` and `alpine` images are throwing errors.
+
+4. Copy needed files from first stage by running
+   `COPY --from=build /go/bin/dlv /`
+
+5. To run Delve during container start change entrypoint of container to:
+
+   ```dockerfile
+   CMD ["/dlv", "--listen=:40000", "--headless=true", "--api-version=2", "--accept-multiclient", "exec", "/app/main"]
```

Comment:
Maybe it is valuable to mention that there is alternative way of doing that. In docker compose we can add:
```dockerfile
ENTRYPOINT ["/dlv"]
```
and in docker compose file:
```docker-compose
command: --listen=:40000 --headless=true --api-version=2 --accept-multiclient exec /app/main -- --loglevel=trace
```

Reply:
Danger of it getting cluttered, and for those who already know docker-compose will have no value for this. Don't see the apparent benifit of doing it via docker-compose instead of inside the Dockerfile

Could you elaborate? Am I missing something?

Reply:
Nop. It is only alternative way IMO easier to write because without quotes and comas.  Second thing that we could mention is if app under debbuger needs own params it have to be prefixed with `--` as in sample below:

```docker-file
command: --listen=:40000 --headless=true --api-version=2 --accept-multiclient exec /app/main -- --loglevel=trace
```

Reply:
Updated docs. Also removed the images in favor of regular diff code blocks.